src/detect-engine.c

```diff
@@ -2000,55 +2007,6 @@ bool DetectEnginePktInspectionRun(ThreadVars *tv,
     return true;
 }
 
-/**
- * \param data pointer to SigMatchData. Allowed to be NULL.
- */
-static int DetectEnginePktInspectionAppend(Signature *s, InspectionBufferPktInspectFunc Callback,
-        SigMatchData *data, const int list_id)
-{
-    DetectEnginePktInspectionEngine *e = SCCalloc(1, sizeof(*e));
-    if (e == NULL)
-        return -1;
-
-    e->mpm = s->init_data->mpm_sm_list == list_id;
-    DEBUG_VALIDATE_BUG_ON(list_id < 0 || list_id > UINT16_MAX);
-    e->sm_list = (uint16_t)list_id;
-    e->sm_list_base = (uint16_t)list_id;
-    e->v1.Callback = Callback;
-    e->smd = data;
-
-    if (s->pkt_inspect == NULL) {
-        s->pkt_inspect = e;
-    } else {
-        DetectEnginePktInspectionEngine *a = s->pkt_inspect;
-        while (a->next != NULL) {
-            a = a->next;
-        }
-        a->next = e;
-    }
-    return 0;
-}
-
-int DetectEnginePktInspectionSetup(Signature *s)
-{
-    /* only handle PMATCH here if we're not an app inspect rule */
-    if (s->sm_arrays[DETECT_SM_LIST_PMATCH] && (s->init_data->init_flags & SIG_FLAG_INIT_STATE_MATCH) == 0) {
```

Comment:
What about the condition `(s->init_data->init_flags & SIG_FLAG_INIT_STATE_MATCH) == 0` ? 
It is no longer here... Can you explain why ?

Reply:
The init_data wouldn't be around anymore by the time of DetectEnginePktInspectionRun(), but shouldn't that check be covered by the following?
https://github.com/coledishington/suricata/blob/detect-inline-pkt-inspect-engines-v5/src/detect-engine-build.c#L1896